### PR TITLE
Refactor normalizer in separate files

### LINKED ISSUE #3290: Split normalizer into smaller building blocks
## Overview
Current version of normalizer is written in one file with case analysis of AST. The file has almost 1500 lines of code and in some places deeply nested.
https://github.com/tgrospic/rchain/blob/825e7c4bc3/rholang/src/main/scala/coop/rchain/rholang/interpreter/compiler/normalize.scala#L1495

### Design
Normalizer file should be split into multiple parts with extracted reusable helper functions.
The goal should be to make it more clear where ambiguous syntax is erased, which part contains source transformation and overall to be more clear what syntax rules are applied.
Also it should be future extendable for debugging support or building source maps between different Rholang versions.

### PR DESCRIPTION
**Pull request by zsluedem**
## Overview

https://github.com/rchain/rchain/issues/3290

Currently this pr only split the big file https://github.com/tgrospic/rchain/blob/825e7c4bc3/rholang/src/main/scala/coop/rchain/rholang/interpreter/compiler/normalize.scala#L1495 into serveral files.

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

File: rholang/src/main/scala/coop/rchain/rholang/interpreter/compiler/normalizer/procNormalizeMatcher/PBundleNormalizer.scala [hunk 1]
```diff
@@ -0,0 +1,73 @@
+package coop.rchain.rholang.interpreter.compiler.normalizer
```
Comment (nzpr) [inline rholang/src/main/scala/coop/rchain/rholang/interpreter/compiler/normalizer/procNormalizeMatcher/PBundleNormalizer.scala]: package here does not match path, and maybe its better to make `procNormalizerMatcher` -> `processes`? As it is in `normalizer` package anyway

**Review by nzpr — APPROVED** [review]
LGTM

**Review by tgrospic — APPROVED** [review]
This is exactly what I had in mind. Thanks!

**Comment by zsluedem** [comment]
bors merge

**Pull request merged**